tkg/clusterclient/clusterclient.go

```diff
@@ -2485,36 +2504,41 @@ func (c *client) UpdateAWSCNIIngressRules(clusterName, clusterNamespace string)
 	}
 
 	cniIngressRules := awsCluster.Spec.NetworkSpec.CNI.CNIIngressRules
-	// first check if existing ingress rules contains the kapp-controller port
-	for _, ingressRule := range cniIngressRules {
-		if ingressRule.Description != "kapp-controller" {
-			continue
-		}
-
-		if ingressRule.Protocol != capav1beta2.SecurityGroupProtocolTCP {
-			continue
+	cniIngressRulesNeedUpdate := false
+	for _, ingressRule := range AWSIngressRules {
+		if !containsIngressRule(cniIngressRules, ingressRule) {
+			cniIngressRules = append(cniIngressRules, ingressRule)
+			cniIngressRulesNeedUpdate = true
 		}
+	}
 
-		if ingressRule.FromPort != DefaultKappControllerHostPort || ingressRule.ToPort != DefaultKappControllerHostPort {
-			continue
+	if cniIngressRulesNeedUpdate {
+		awsCluster.Spec.NetworkSpec.CNI.CNIIngressRules = cniIngressRules
+		if err := c.UpdateResource(awsCluster, clusterName, clusterNamespace); err != nil {
```

Comment:
Is it better to use UpdateResourceWithPolling or PatchResource with pollOptions
or maybe
```
import "k8s.io/client-go/util/retry"

err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
		return clusterClient.UpdateResource(awsCluster, clusterName, clusterNamespace)
	})
```
to make it more robust to prevent the error below
```
 the object has been modified; please apply your changes to the latest version and try again
```

Reply:
done